src/Monad/Writer/functions.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Widmogrod\Monad\Writer;
+
+use Widmogrod\Monad as M;
+
+const pure = 'Widmogrod\Monad\Writer\pure';
+
+/**
+ * pure :: Applicative Just f => a -> f a
+ *
+ * @param mixed $f
+ * @return M\Writer
+ */
+function pure($f)
+{
+    return M\Writer::of($f);
+}
+
+const value = 'Widmogrod\Monad\Writer\value';
+
+/**
+ * value :: Writer w m => a -> m a
+ *
+ * Replace the value inside the monad
+ *
+ * @param mixed $value
+ * @return callable
+ */
+function value($value)
+{
+    return function() use ($value) {
+        return M\Writer::of($value);
```

Comment:
In this case it would be exactly the same as `pure`.

Honestly I am not sure what this function is supposed to return, I took it from your `State` implementation. From what I understood the idea is to create a function, that if bound using `bind` changes the value inside the Monad.

If I understood correctly, we need to return a function.

Did I get it wrong ?